pkg/kubelet/kubelet_pods.go

```diff
@@ -617,6 +617,13 @@ func (kl *Kubelet) podFieldSelectorRuntimeValue(fs *v1.ObjectFieldSelector, pod
 		return pod.Spec.NodeName, nil
 	case "spec.serviceAccountName":
 		return pod.Spec.ServiceAccountName, nil
+	case "status.hostIP":
+		hostIP, err := kl.GetHostIP()
```

Comment:
Use `kl.getHostIPAnyWay()` to make sure we can always get the host IP, even if kubelet hasn't registered yet.